src/lib/SideBar.svelte

```diff
@@ -21,15 +20,46 @@
 		{ icon: 'fab fa-discord', info: 'Discord', activity: null },
 		{ icon: 'fab fa-github', info: 'GitHub', activity: null },
 	];
-
-	let activeInfo = null;
-
+	let activeInfo = null; // Tracks currently visible info
+	let lastHoveredInfo = null; // Tracks last hovered info
+	let isPanelHovered = false; // Tracks if info panel is hovered
+	let hideTimeout = 0; // Timeout for hiding info panel
 	function showInfo(info) {
-		activeInfo = info;
+		if (hideTimeout) {
+			hideTimeout = 0;
+		}
+		if (info !== lastHoveredInfo) {
+			activeInfo = info;
+			lastHoveredInfo = info;
+		}
 	}
-
 	function hideInfo() {
-		activeInfo = null;
+		if (!hideTimeout) {
+			hideTimeout = setTimeout(() => {
+				if (!isPanelHovered) {
```

Comment:
I still think this logic is redundant. Below we should call clear timeout so that the callback is never even called if isPanelHovered is set to true.